### PR TITLE
added python solution to Milk Pails (620)

### PR DESCRIPTION
**Pull request by Amaryllidaceae**
_If any of the below don't apply to this Pull Request, mark the checkbox as done._

- [X] I have tested my code.
- [X] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [X] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which include the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making the changes.

### CODE REVIEW

File: solutions/usaco-620.mdx [hunk 1]
```diff
@@ -332,4 +332,48 @@ public class pails
...
+def generate_permutations(p1: int, p2: int):
+	# generates results of all possible pours
+	da = min(p2, x-p1) # amount poured when pouring p2 into p1
+	db = min(p1, y-p2) # amount poured when pouring p1 into p2
+	return [(x, p2), (p1, y), (0, p2), (p1, 0), (p1 + da, p2 - da),
+			(p1 - db, p2 + db)]
+
+x, y, k, m = map(int, input().split())
+
+dist = [[-1]*(y+1) for _ in range(x+1)]
```
Comment (SansPapyrus683) [inline solutions/usaco-620.mdx:352]: spaces between operators

File: solutions/usaco-620.mdx [hunk 2]
```diff
@@ -332,4 +332,48 @@ public class pails
...
+import sys
+sys.stdin, sys.stdout = open('pails.in', 'r'), open('pails.out', 'w')
+input = sys.stdin.readline
+
+def generate_permutations(p1: int, p2: int):
+	# generates results of all possible pours
+	da = min(p2, x-p1) # amount poured when pouring p2 into p1
+	db = min(p1, y-p2) # amount poured when pouring p1 into p2
+	return [(x, p2), (p1, y), (0, p2), (p1, 0), (p1 + da, p2 - da),
+			(p1 - db, p2 + db)]
```
Comment (Aadit-Ambadkar) [inline solutions/usaco-620.mdx:348]:
```suggestion
	return [
		(x, p2), 
		(p1, y), 
		(0, p2), 
		(p1, 0), 
		(p1 + da, p2 - da),
		(p1 - db, p2 + db)
	]
```

**Review by Aadit-Ambadkar — APPROVED** [review]
nice work!

File: solutions/usaco-620.mdx (hunk 1 above)
Comment (SansPapyrus683) [reply solutions/usaco-620.mdx]: not addressed

Comment (Amaryllidaceae) [reply solutions/usaco-620.mdx]: think I fixed it now

File: solutions/usaco-620.mdx [hunk 3]
```diff
@@ -332,4 +332,54 @@ public class pails
...
+		(p1 + da, p2 - da),
+		(p1 - db, p2 + db)
+	]
+
+x, y, k, m = map(int, input().split())
+
+dist = [[-1] * (y + 1) for _ in range(x + 1)]
+dist[0][0] = 0
+
+min_diff = 201
```
Comment (SansPapyrus683) [inline solutions/usaco-620.mdx:361]: make this `float('inf')`

Comment (Amaryllidaceae) [reply solutions/usaco-620.mdx]: done

File: solutions/usaco-620.mdx [hunk 4]
```diff
@@ -332,4 +332,54 @@ public class pails
...
+```python
+from collections import deque
+import sys
+sys.stdin, sys.stdout = open('pails.in', 'r'), open('pails.out', 'w')
+input = sys.stdin.readline
+
+def generate_permutations(p1: int, p2: int):
+	# generates results of all possible pours
+	da = min(p2, x - p1) # amount poured when pouring p2 into p1
+	db = min(p1, y - p2) # amount poured when pouring p1 into p2
```
Comment (SansPapyrus683) [inline solutions/usaco-620.mdx:346]:
```suggestion
	da = min(p2, x - p1)  # amount poured when pouring p2 into p1
	db = min(p1, y - p2)  # amount poured when pouring p1 into p2
```

**Pull request merged**